### ISSUE TITLE
the stage of execution too slow

### ISSUE THREAD

**Issue by SuperEdison** [node-real/bsc-erigon#8]
This should only be used in very rare cases e.g. if you are not 100% sure if something is a bug or asking a question that leads to improving the documentation. For general questions please use [Erigon's discord](https://discord.gg/hQ2dxbNZ).
![image](https://user-images.githubusercontent.com/84554944/226095144-c5cfbbf0-9248-4bb8-b27e-a6c4a9dd545f.png)
blk only 2-4, forever can't catch up lastest block

**Comment by calmbeing** [comment]
Stuck at 26605878? Can you provide more logs

**Comment by calmbeing** [comment]
Because of execution time, execution stage is the most time-consuming stage within all sync stages of Erigon, suggest download mdbx snapshot and restart, it'll be faster. Also, in the future, it'll be possible to download state from bit-torrrent instead of execute locally.

**Comment by calmbeing** [comment]
> > Because of execution time, execution stage is the most time-consuming stage within all sync stages of Erigon, suggest download mdbx snapshot and restart, it'll be faster. Also, in the future, it'll be possible to download state from bit-torrrent instead of execute locally.
> 
> you mean delete chaindata mdbx.dat keep snapshots file?and restart erigon?

Download the mdbx.dat with full archive data and replace the current chaindata/mdbx.dat, then restart erigon.

**Comment by ty-sentio-xyz** [comment]
You might be interested in trying this out: https://github.com/sentioxyz/erigon

This is a custom fork from sentio.xyz, the motivation came from the heavy utilization of cloud disks in our production.

Cloud disks provide same throughout but much higher latency comparing to a local NVMe disk.  The current design of erigon is not really friendly to cloud disks, as most sync stages are only leveraging sequential I/O (which is very slow: https://cloud.google.com/compute/docs/disks/optimizing-pd-performance#io-queue-depth).  

So we decided to introduce parallel execution within selected stages, to fully utilize I/O queue of cloud disks.
These are the main improvements:

- HashState Stage: Parallel data extraction.  This improves performance when page cache misses.

- Intermediate Hash Stage: Parallel trie hash calculation, by calculating the hash of 16 subtrees then merge.  This could improve speed of that stage significantly.

- LogIndex/CallTraceIndex/AccountHistory/StorageHistory:  These stages tend to be very slow when updating bitmaps.  Bitmaps need to be read from mdbx and rewritten, all sequential.  We've changed that stage to parallel extract data into collector, then designed another database (bitmapdb2) to offload bitmaps from mdbx.  (This requires `--bitmapdb2` flag.  Once turned on, you cannot use a separated rpcdaemon, as bitmapdb2 cannot be opened by multiple processes.  Also you cannot enable otterscan APIs, as the new database do not yet support these.)

Issues/Criticism/PRs are welcome.
